### PR TITLE
Improve Outputs When Focusing in Eng Diff 2 GUI

### LINKED ISSUE #28360: Engineering Interface Focusing Outputs
**Original reporter:** Saurabh (ISIS)
 
The output from the focusing of data files should include a Topas compatible output (`.abc` is what Saurabh has been using). Should output (`.nxs`, `.gss`, and `.abc`)

A text file containing the averaged logs for that run should also be included in the set of output files.

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
- Replaced the output of `.dat` files with TOPAS compatible `.abc` files. 
- Made focusing output averaged values of all numerical sample logs into a CSV file. 

**To test:**
1. Interfaces -> Diffraction -> Engineering Diffraction
1. Load or create a Calibration V#: 307521 CeO2#: 305738
2. Set an RB number
2. Focus some run files. Anything in the range 305500-308000 should work. 
3. Check that for each bank and run there is a `.gss`, `.nxs`, and `.abc` file. 
4. Check that for each run there is a sample log `.csv`

Fixes #28360 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by StephenSmith25 — COMMENTED** [review]
Functions as described. Just a few comments

File: scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py [hunk 1]
```diff
@@ -122,6 +133,36 @@ def test_save_output_files_with_RB_number(self, nexus, gss, xye):
...
+    @patch(file_path + ".csv")
+    def test_output_sample_logs_with_rb_number(self, mock_csv, mock_path, mock_logger):
+        mock_writer = MagicMock()
+        mock_csv.writer.return_value = mock_writer
+        mock_path.return_value = self.test_dir
+        ws = CreateSampleWorkspace()
+
+        self.model._output_sample_logs("ENGINX", "00000", ws, "0")
+
+        self.assertEqual(5, len(ws.getRun().keys()))
```
[the remark this answers is not in the document]
Comment (ConorMFinn) [reply scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py]: You're right, they aren't really testing anything. Not sure what my thinking was there.

File: scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py [hunk 2]
```diff
@@ -154,19 +157,53 @@ def _save_focused_output_files_as_nexus(self, instrument, sample_path, bank, sam
...
                            Filename=xye_output_path,
-                           SplitFiles=False)
+                           SplitFiles=False,
+                           Format="TOPAS")
+
+    @staticmethod
+    def _output_sample_logs(instrument, run_number, workspace, rb_num):
+        def write_to_file():
+            with open(output_path, "w", newline="") as logfile:
+                writer = csv.writer(f, ["Sample Log", "Avg Value"])
```
Comment (StephenSmith25) [inline scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py]: Sorry, was thinking at the time that I should have commented both - but this also needs to be changed to logfile 
```suggestion
                writer = csv.writer(logfile, ["Sample Log", "Avg Value"])
```

Comment (ConorMFinn) [reply scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py]: Yep, my bad. Should've checked. Fixed now.

**Comment by StephenSmith25** [comment]
I'm happy with this provided the builds pass.

**Review by StephenSmith25 — APPROVED** [review]

**Pull request merged**